sdk/lib/opentelemetry/sdk/trace/tracer.rb

```diff
@@ -122,6 +119,18 @@ def create_link(span_context, attrs = nil)
           @active_trace_config.trim_link_attributes(attrs)
           Link.new(span_context: span_context, attributes: attrs)
         end
+
+        private
+
+        def internal_create_span(result, name, kind, trace_id, span_id, parent_span_id, attributes, links, events, start_timestamp)
+          if result.record_events? && !@stopped
+            trace_flags = result.sampled? ? OpenTelemetry::Trace::TraceFlags::SAMPLED : OpenTelemetry::Trace::TraceFlags::DEFAULT
+            context = OpenTelemetry::Trace::SpanContext.new(trace_id: trace_id, trace_flags: trace_flags)
+            Span.new(context, name, kind, parent_span_id, @active_trace_config, @active_span_processor, attributes, links, events, start_timestamp || Time.now)
```

Comment:
My interpretation of the sampling [OTEP](https://github.com/open-telemetry/oteps/blob/master/text/0006-sampling.md#sampler-interface) is that the initial set of attributes are the ones passed to start span, and the additional attributes, to be added later, are those returned on the sampling result.

If that interpretation is correct, then we probably need to merge the attributes from the `SamplingResult` with the initial attributes here.

Reply:
Good catch, thanks! I’ll fix that up tomorrow and add some tests.

Reply:
Fixed